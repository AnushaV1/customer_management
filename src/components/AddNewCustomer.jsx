import React from "react";
import { Formik, Form, Field, ErrorMessage } from "formik";
import 'bootstrap/dist/css/bootstrap.min.css';
import * as Yup from "yup";
import CustomerAPI from "../CustomerAPI";

const AddNewCustomer = () => {

    const initialValues = {
        firstname: '',
        lastname:  '',
        address: '',
        city: '',
        state:'',
        email:'',
        phone:''
    }
    
    const validationSchema = Yup.object({
        firstname: Yup.string().required('Required!'),
        lastname: Yup.string().required('Required!'),
        address: Yup.string().required('Required!'),
        city: Yup.string().required('Required!'),
        state:Yup.string().required('Required!'),
        phone:Yup.number().required('Required!'),
        email: Yup.string()
            .email('Invalid email format')
            .required('Required!')
    })

    const onSubmit = async(values) => {
        //console.log(values);
        try {
            const res = await CustomerAPI.AddNewCustomer(values);

        }
        catch(error) {
            console.log(error);
        }
    }
    
    return (
        <div className="container">
        <div className="row">
        <div className="col-lg-6 offset-lg-3 form-wrapper">
        <h2>Add New Customers</h2>
        <Formik initialValues={initialValues} validationSchema= {validationSchema} onSubmit={onSubmit}>
            <Form>
                <div className="form-group">
                <label htmlFor="firstname">First Name</label>
                <Field type="text" name="firstname"
                    placeholder="First Name" 
                    className='form-control'
                />
                <ErrorMessage name ="firstname" render={msg => <div className="error">{msg}</div>} />
            
            </div>
            <div className="form-group">
                <label htmlFor="lastname">Last Name</label>
                <Field type="text" name="lastname"  placeholder="Last Name"
                    className='form-control' />
                    <ErrorMessage name ="lastname"  render={msg => <div className="error">{msg}</div>} />
            </div>
            <div className="form-group">
                <label htmlFor="username">Address</label>
                <Field type="text" name="address"  placeholder="Address"
                    className='form-control'
                />
                <ErrorMessage name ="address" render={msg => <div className="error">{msg}</div>} />
            
            </div>
        <div className="form-group">
        <label htmlFor="state">State</label>
        <Field as="select" name="state" placeholder="State"
            className='form-control'>
            <option value="California" default>California</option>
            <option value="Texas">Texas</option>
            <option value="Florida">Florida</option>
            <option value="Washington">Washington</option>
            </Field>
            <ErrorMessage name ="state" render={msg => <div className="error">{msg}</div>} />
        </div>
            <div className="form-group">
                <label htmlFor="city">City</label>
                <Field as="select" name="city" placeholder="City"
                    className='form-control'>
                    <option value="San Jose" default>San Jose</option>
                    <option value="Cupertino">Cupertino</option>
                    <option value="Los Angeles">Los Angeles</option>
                    <option value="San Diego">San Diego</option>
                    </Field>
                    <ErrorMessage name ="city" render={msg => <div className="error">{msg}</div>} />
            </div>
            <div className="form-group">
            <label htmlFor="email">Email</label>
            <Field type="text" name="email"
                    placeholder="Email" className='form-control'
                />
                <ErrorMessage name ="email" render={msg => <div className="error">{msg}</div>} />
            </div>
            <div className="form-group">
            <label htmlFor="phone">Phone</label>
            <Field type="phone" name="phone"
                    placeholder="Phone" className='form-control'
                />
                <ErrorMessage name ="phone" render={msg => <div className="error">{msg}</div>} />
            </div>
                    
            <div className="form-group">
                <button type="submit" className="btn btn-primary">
            
                    Add User
                </button>{' '}
                <button type="reset" className="btn btn-secondary">Reset</button>
                </div>
        </Form>
        </Formik>
    </div>
    </div>
    </div>
    )
}

export default AddNewCustomer;
