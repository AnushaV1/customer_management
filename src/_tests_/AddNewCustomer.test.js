import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import AddNewCustomer from '../components/AddNewCustomer';


it('renders without crashing', function () {
	render(
			<BrowserRouter>
				<AddNewCustomer />
			</BrowserRouter>
	);
});

it('matches snapshot', function () {
	const { asFragment } = render(
			<BrowserRouter>
				<AddNewCustomer />
			</BrowserRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});