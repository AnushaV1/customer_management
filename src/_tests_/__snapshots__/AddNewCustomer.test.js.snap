// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-6 offset-lg-3 form-wrapper"
      >
        <h2>
          Add New Customers
        </h2>
        <form
          action="#"
        >
          <div
            class="form-group"
          >
            <label
              for="firstname"
            >
              First Name
            </label>
            <input
              class="form-control"
              name="firstname"
              placeholder="First Name"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="lastname"
            >
              Last Name
            </label>
            <input
              class="form-control"
              name="lastname"
              placeholder="Last Name"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="username"
            >
              Address
            </label>
            <input
              class="form-control"
              name="address"
              placeholder="Address"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="state"
            >
              State
            </label>
            <select
              class="form-control"
              name="state"
              placeholder="State"
            >
              <option
                default=""
                value="California"
              >
                California
              </option>
              <option
                value="Texas"
              >
                Texas
              </option>
              <option
                value="Florida"
              >
                Florida
              </option>
              <option
                value="Washington"
              >
                Washington
              </option>
            </select>
          </div>
          <div
            class="form-group"
          >
            <label
              for="city"
            >
              City
            </label>
            <select
              class="form-control"
              name="city"
              placeholder="City"
            >
              <option
                default=""
                value="San Jose"
              >
                San Jose
              </option>
              <option
                value="Cupertino"
              >
                Cupertino
              </option>
              <option
                value="Los Angeles"
              >
                Los Angeles
              </option>
              <option
                value="San Diego"
              >
                San Diego
              </option>
            </select>
          </div>
          <div
            class="form-group"
          >
            <label
              for="email"
            >
              Email
            </label>
            <input
              class="form-control"
              name="email"
              placeholder="Email"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="phone"
            >
              Phone
            </label>
            <input
              class="form-control"
              name="phone"
              placeholder="Phone"
              type="phone"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <button
              class="btn btn-primary"
              type="submit"
            >
              Add User
            </button>
             
            <button
              class="btn btn-secondary"
              type="reset"
            >
              Reset
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
